src/sentry/integrations/vsts/integration.py

```diff
@@ -463,8 +468,36 @@ def build_integration(self, state: Mapping[str, Any]) -> Mapping[str, Any]:
             integration_model = IntegrationModel.objects.get(
                 provider="vsts", external_id=account["accountId"], status=ObjectStatus.ACTIVE
             )
-            # preserve previously created subscription information
-            integration["metadata"]["subscription"] = integration_model.metadata["subscription"]
+
+            # Get Integration Metadata
+            integration_migrated = integration_model.metadata.get("integration_migrated", False)
+            if (
+                features.has(
+                    "organizations:migrate-azure-devops-integration", self.pipeline.organization.id
+                )
+                and not integration_migrated
+            ):
+                subscription_id, subscription_secret = self.create_subscription(
+                    base_url=base_url, oauth_data=oauth_data
+                )
+                integration["metadata"]["subscription"] = {
+                    "id": subscription_id,
+                    "secret": subscription_secret,
+                }
+
+                integration["metadata"]["integration_migrated"] = True
+
+                logger.info(
+                    "vsts.build_integration.migrated",
+                    extra={
+                        "organization_id": self.pipeline.organization.id,
+                        "user_id": user["id"],
+                        "account": account,
```

Comment:
Do we want to log the subscription Id here as well?

Reply:
old integration id too? Just to help debugging if someone has multiple Azure